cwms-data-api/src/test/java/cwms/cda/data/dao/MeasurementDaoTest.java

```diff
@@ -0,0 +1,238 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import usace.cwms.db.jooq.codegen.udt.records.LOCATION_REF_T;
+import usace.cwms.db.jooq.codegen.udt.records.STREAMFLOW_MEAS2_T;
+import usace.cwms.db.jooq.codegen.udt.records.SUPP_STREAMFLOW_MEAS_T;
+
+final class MeasurementDaoTest {
+    @Test
+    void testFromJooqMeasurementRecord() {
+        Instant instant = Instant.parse("2024-01-01T00:00:00Z");
+        STREAMFLOW_MEAS2_T record = mock(STREAMFLOW_MEAS2_T.class);
+        when(record.getLOCATION()).thenReturn(mockLocation());
+        when(record.getMEAS_NUMBER()).thenReturn("12345");
+        when(record.getAGENCY_ID()).thenReturn("USGS");
+        when(record.getPARTY()).thenReturn("SomeParty");
+        when(record.getDATE_TIME()).thenReturn(Timestamp.from(instant));
+        when(record.getFLOW()).thenReturn(100.0);
+        when(record.getGAGE_HEIGHT()).thenReturn(2.0);
+        when(record.getQUALITY()).thenReturn("good");
+        when(record.getWM_COMMENTS()).thenReturn("Test comment");
+        when(record.getAREA_UNIT()).thenReturn("ft2");
+        when(record.getFLOW_UNIT()).thenReturn("cfs");
+        when(record.getHEIGHT_UNIT()).thenReturn("ft");
+        when(record.getVELOCITY_UNIT()).thenReturn("fps");
+        when(record.getTEMP_UNIT()).thenReturn("F");
+        when(record.getUSED()).thenReturn(JooqDao.formatBool(true));
+        when(record.getAIR_TEMP()).thenReturn(25.0);
+        when(record.getCUR_RATING_NUM()).thenReturn("1");
+        when(record.getCTRL_COND_ID()).thenReturn("UNSPECIFIED");
+        when(record.getFLOW_ADJ_ID()).thenReturn("UNKNOWN");
+        when(record.getDELTA_HEIGHT()).thenReturn(0.5);
+        when(record.getDELTA_TIME()).thenReturn(60.0);
+        when(record.getPCT_DIFF()).thenReturn(10.0);
+        when(record.getREMARKS()).thenReturn("Some remarks");
+        when(record.getSHIFT_USED()).thenReturn(11.0);
+        when(record.getWATER_TEMP()).thenReturn(15.0);
+        when(record.getSUPP_STREAMFLOW_MEAS()).thenReturn(mockSupplementalStreamflowMeasurement());
+
+        Measurement measurement = MeasurementDao.fromJooqMeasurementRecord(record);
+
+        assertNotNull(measurement);
+        assertEquals("12345", measurement.getNumber());
+        assertEquals("USGS", measurement.getAgency());
+        assertEquals("SomeParty", measurement.getParty());
+        assertEquals(instant, measurement.getInstant());
+        assertEquals("Test comment", measurement.getWmComments());
+        assertEquals("ft2", measurement.getAreaUnit());
+        assertEquals("cfs", measurement.getFlowUnit());
+        assertEquals("ft", measurement.getHeightUnit());
+        assertEquals("fps", measurement.getVelocityUnit());
+        assertEquals("F", measurement.getTempUnit());
+        assertTrue(measurement.isUsed());
+
+        //Assertions for Streamflow meas fields
+        assertEquals(100.0, measurement.getStreamflowMeasurement().getFlow());
```

Comment:
Use assertEquals for doubles where it includes a fudge factor.  The fudge factor should be pretty small, like 0.001.  This applies to all your double assertions.

Reply:
right, missed that in the unit tests. Updated